[docs] Update Learn page copy and resource list (#31989)
mui/material-ui: docs/data/material/getting-started/learn/learn.md

# Learning resources

<p class="description">New to Material UI? Get up to speed quickly with our curated list of learning resources.</p>

## Your first component

The following code snippet demonstrates a basic Material UI app that features a `<Button>` component:

```jsx
import * as React from 'react';
import ReactDOM from 'react-dom';
import Button from '@mui/material/Button';

function App() {
  return (
    <Button variant="contained" color="primary">
      Hello World
    </Button>
  );
}

ReactDOM.render(<App />, document.querySelector('#app'));
```

In the interactive demo below, try changing the code and see how it affects the output. (Hint: change `variant` to `"outlined"` and `color` to `"secondary"`. For more options, see the [`Button` component page](components/buttons/).)

> {{"demo": "../usage/Usage.js", "hideToolbar": true, "bg": true}}

Reviewer: **Off-topic**.

This .. sounds like a good signal that restoring absolute path imports would be better.

Reviewer 2: Do you have a discussion opened about this topic ?
Because in most demos, the new relative path is much simpler to play with that the absolute path we had before.

Reviewer: Interesting feedback. I can only recall another comment I left on a PR of Jun but I never followed what was the answers 😁. The problem I envision is about being able to quickly find a demo by looking at the markdown. Before, I could open the markdown, and copy and paste the path in my IDE, it would at 100% return the right file. Now, we have relative paths and non-unique demo files. It's trickier to open the right file.

I have no visibility on the upside of this new approach, I also so the cons, I guess it's a tradeoff then.

Reviewer 2: On X the demo is always on the same demo as the markdown
So it is the opposite, the relative path makes it very obvious and help me find the demo quicker.

Reviewer 3: No preference on my side. What relative path really benefits me is the migration when I move them into different folders (it just works).

> quickly find a demo source by looking at the markdown

Usually, I copy the last part of the demo and search. I haven't encountered the wrong file before because most demos have a unique name based on the folder/component name.

Reviewer: It sounds like the pain I'm referring to doesn't resonate, fair enough. I think that we can pause it here. I leave this comment unresolved for visibility in case it comes up in the future.

> # Learning Resources

Reviewer: ```suggestion
# Learning resources
```

Author: "Learn" on the side nav is kind of vague. Should we change that to "Learning Resources"?

Reviewer 4: "Learning resources" used on both the sidebar and title sounds great to me 👍

<p class="description">New to Material UI? Get up to speed quickly with our curated list of learning resources.</p>

## Your first component

> The following code snippet contains everything you need to start building an app with Material UI, beginning with the `<Button>` component:

Reviewer 5: > contains everything you need

I wonder... does it?! Ever since I read this line (it has been on the documentation for quite some time) it got me thinking. I assume that "everything" refers to the component import and its usage inside a function. For some reason, it doesn't feel like "everything".

Reviewer 3: I think it is exaggerated. It can be something like: `Let's start with a Button!`

Author: Good call! "Everything you need" is a bit of an overstatement. I'll come up with a different way to introduce this code snippet instead.

> The following code snippet contains everything you need to start building an app with Material UI, beginning with the `<Button>` component:

Author: ```suggestion
The following code snippet demonstrates a basic Material UI app that features a `<Button>` component:
```

Reviewer 3: Perfect!

```jsx
import * as React from 'react';
import ReactDOM from 'react-dom';
import Button from '@mui/material/Button';

function App() {
  return (
    <Button variant="contained" color="primary">
      Hello World
    </Button>
  );
}

ReactDOM.render(<App />, document.querySelector('#app'));
```

> In the interactive demo below, try changing the code and see how it affects the output. (Hint: change `variant` to `"outlined"` and `color` to `"secondary"`. For more options, see the [`Button` component page](components/buttons/).)

Reviewer: ```suggestion
In the interactive demo below, try changing the code and see how it affects the output. (Hint: change `variant` to `"outlined"` and `color` to `"secondary"`. For more options, see the [`Button` component page](/components/buttons/).
```

Author: The last period there should be inside of the parentheses, since it's a complete sentence that stands on its own (as opposed to an additional clause, like this example 😁).

Reviewer 4: Let's trust Sam he is native :D. It is the same in my language so it makes sense :)

{{"demo": "../usage/Usage.js", "hideToolbar": true, "bg": true}}

## Example projects

Visit the [example projects](/getting-started/example-projects/) page to see how we recommend implementing Material UI with various React libraries and frameworks like Next.js, Gatsby, Create React App, and more.

## Templates

Check out our [selection of basic templates](/getting-started/templates/) to get started building your next app more quickly.

## Recommended resources

Beyond our official documentation, there are countless members of our community who create fantastic tutorials and guides for working with Material UI.

The following is a curated list of some of the best third-party resources we've found—both free and paid—for learning how to build beautiful apps with our components.

> > **Note:** as of Spring 2022, the paid resources available all refer to outdated versions of Material UI (before v5) so unless you're using an older version, you will be more successful with the free resources.

Reviewer 5: ```suggestion
> **Note:** as of the second quarter of 2022, the paid resources available all refer to outdated versions of Material UI (before v5) so unless you're using an older version, you will be more successful with the free resources.
```
Making it not too Northern Hemisphere centric 😅

Author: Thank you!!

- [**Next.js 11 Setup with Material UI v5**](https://www.youtube.com/watch?v=IFaFFmPYyMI) by Leo Roese: learn how to integrate Material UI into your Next.js app, using Emotion as the style engine.

- [**Create a Responsive Navigation Bar with Material UI v5**](https://www.youtube.com/watch?v=lUkxSnJ7aDw) by Nikhil Thadani: detailed walkthrough using Material UI with Create React App.

- [**The Clever Dev**](https://www.youtube.com/channel/UCb6AZy0_D1y661PMZck3jOw) and [**The Smart Devpreneur**](https://smartdevpreneur.com/category/javascript/material-ui/) by Jon M: dozens of high-quality videos and articles digging deep into the nuts and bolts of Material UI.

### Paid

- [**Implement High Fidelity Designs with Material-UI and ReactJS**](https://www.udemy.com/course/implement-high-fidelity-designs-with-material-ui-and-reactjs/) course by Zachary Reese: Bridge the gap between design & development. Break down detailed designs and bring them to life with Material UI (v4) and React.

> - [**Material UI with React**](https://bonsaiilabs.com/courseDetail/material-ui-with-react/) course by bonsaiilabs: Learn the fundamentals of Google Material Design and how to develop an end-to-end flight search and booking application using Material UI and React.

Reviewer: This course never really took up, I would remove it, it's also on v4.

Author: Should we consider removing the Paid resources section entirely? The two remaining courses appear to be in the process of updating to v5, but I don't know when we can expect them to be released. We could remove the section for now and replace it when new content is available.

Reviewer 4: Sam agree, let's comment out these section and add a comment to re-visit this in maybe Q3 and check if we should add those back in. We would also remove the Note we have above the free resources.

> - [**React Material UI Cookbook**](https://www.amazon.com/gp/product/1789615224/) by Adam Boduch: Build modern-day applications by implementing Material Design principles in React, using Material UI.

Reviewer: It's 3 years old, I would remove this.
